xdsl/dialects/vector.py

```diff
@@ -292,6 +302,78 @@ def get(mask_operands: list[Operation | SSAValue]) -> CreatemaskOp:
         )
 
 
+@irdl_op_definition
+class TransferReadOp(IRDLOperation):
+    name = "vector.transfer_read"
+
+    source = operand_def(TensorOrMemrefOf(Attribute))
+    indices = var_operand_def(IndexType)
+    padding = operand_def(Attribute)
+    mask = opt_operand_def(VectorType[I1])
+
+    permutation_map = prop_def(AffineMapAttr)
+    in_bounds = opt_prop_def(ArrayAttr[BoolAttr])
+
+    result = result_def(VectorType)
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True)]
+
+    def verify_(self):
+        assert isa(self.source.type, MemRefType[Attribute] | TensorType[Attribute])
+        assert isa(self.result.type, VectorType[Attribute])
+        # TODO verify.
+
```

Comment:
Verification might be a lot of work, but maybe you could add parts of it?
Checking rank, element type and mask should not be that bad, no?

Once this is merged, could you please open an issue to track what is missing?